audio_program_generator/apg.py

```diff
@@ -68,14 +69,22 @@
 from tqdm import tqdm
 
 
-def num_lines_in_file(filename: Path) -> int:
-    """
-    Takes text filename ; returns the number of lines in a file
-    """
-    numlines = 1
-    with open(filename, "r"):
-        numlines += 1
-    return numlines
+def parse_textfile(filename: str="") -> list:
+    def clean(input: str="") -> str:
```

Comment:
Any reason to nest functions? Decorators crossed my mind ...

Reply:
As discussed in meeting, this is a 2-pass regex-based function to clean up user input and then extract ("capture") text phrases and durations. 

First is a negated character class, which excludes all the characters we don't want:
`[^A-Za-z0-9\s;\v]`
and then we sub() all of those with nothing `''`.

Next is a regex that anchors at beginning and end of line (to match whole lines), spaces are outside the capture groups, and the first capture group is non-greedy to not capture the spaces on either side:
`^\s*([\w\s]+?)\s*;\s*(\d+)\s*$`

Reply:
Cool, love regexes, very powerful!